src/sentry/utils/pubsub.py

```diff
@@ -58,15 +47,25 @@ def worker():
         self._started = True
         return True
 
-    def put(self, channel, data):
+    def publish(self, channel, item):
         if not self._start():
             return
 
-        sample_channel = getattr(settings, 'PUBSUB_SAMPLING', {}).get(channel, 1.0)
+        sample_channel = getattr(settings, 'PUBSUB_SAMPLING', 1.0)
         if random.random() <= sample_channel:
             try:
-                self.q.put((channel, data), block=False)
+                self.q.put((channel, item), block=False)
             except Queue.Full:
                 return
 
-pubsub = PubSub()
+
+class RedisPublisher():
+    rds = None
```

Comment:
Maybe I forgot all my Python, but I don't think this is doing what you expected. This is on the class, I think you just want to do an `else` and set it on `self` in `__init__`?

Reply:
this works, basically because attribute lookup on an instance falls through to the class if it isn't set on the instance. But I can see how it looks wonky. will change.

Reply:
Damn, you're right, guess I'm just not used to using `self` and hitting a class attr.